src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs

```diff
@@ -126,8 +126,13 @@ internal abstract class TexlFunction : IFunction
         /// </summary>
         public virtual bool HasPreciseErrors => false;
 
-        // Returns true if the function will mutate the value of argument 0, as is the case with Patch, Collect, Remove, etc.
-        public virtual bool MutatesArg0 => false;
+        /// <summary>
+        /// Returns true if the function will mutate the value of the first argument, as is the case of Patch, Collect, Remove, etc.
+        /// Set can also mutate, but needs to make a decision based on the argument's node.
+        /// This function covers both CanMutate and CanSetMutate scenarios which is checked in CheckTypes/CheckSemantics.
+        /// </summary>
+        /// <param name="arg"></param>

```

Comment:
Which arg is this? Is it the arg0?

Reply:
I've debated this one.  It is arg0, because this is only called for arg0.  But that is a little unclear, so I decided to add the index to the signature and make it clearer, which also adds slightly more expressiveness (any argument can mutate).